contentcuration/contentcuration/statistics.py

```diff
@@ -0,0 +1,186 @@
+
+            if channel.deleted:
+                action_attributes['action'] = 'Delete'
+            # ricecooker uploads and publish actions result in multiple saves, so we filter here only for human updates
+            # chef_tree only exists on ricecooker uploads
+            # channel's main_tree differs from original_channel's on ricecooker uploads
+            # channel's version differs from original_channel's on publish calls
+            elif not channel.chef_tree and channel.main_tree == original_channel.main_tree \
+                    and channel.version == original_channel.version:
+                action_attributes['action'] = 'Update'
+                action_attributes['action_type'] = 'Metadata'
+    else:
+        action_attributes['content_source'] = 'Human'
+        action_attributes['action_source'] = 'Human'
+
+        if channel.main_tree:
+            action_attributes['channel_num_resources'] = channel.main_tree.get_descendants().exclude(
+                kind=content_kinds.TOPIC).count()
+            action_attributes['channel_num_nodes'] = channel.main_tree.get_descendant_count()
+        else:
+            action_attributes['channel_num_resources'] = 0
+            action_attributes['channel_num_nodes'] = 0
+
+        if original_channel is None:
+            if channel.name:
+                action_attributes['action'] = 'Create'
+        elif channel.deleted:
+            action_attributes['action'] = 'Delete'
+        elif channel.version == original_channel.version:
+            action_attributes['action'] = 'Update'
+            action_attributes['action_type'] = 'Metadata'
+
+    record_channel_action_stats(action_attributes)
+
+
+def record_node_addition_stats(nodes_being_added, original_first_node, user_id):
+    """
+    :param nodes_being_added: The nodes being added to the human channel.
+    :param original_first_node: The original state of the first node being added.
+    :param user_id: The id of the user committing the action.
+    """
+    action_attributes = dict(action_source='Human', content_source='Human', user_id=user_id)
+
+    # The first node to be added in this action.
+    first_node = nodes_being_added.itervalues().next()
+    action_attributes['content_type'] = first_node['kind'].kind.title()
+
+    num_resources = 0
+    if 'id' in first_node and original_first_node.parent is not None:
+        action_attributes['action'] = 'Update'
+    else:
+        action_attributes['action'] = 'Create'
+        action_attributes['num_nodes_added'] = len(nodes_being_added)
+        for id, node in nodes_being_added.items():
+            if node['kind'].kind != content_kinds.TOPIC:
+                num_resources += 1
+        action_attributes['num_resources_added'] = num_resources
+
+    # The parent the new nodes are being added to.
+    parent_node = first_node['parent']
+    action_attributes['channel_id'] = parent_node.get_channel().id
```

Comment:
Need to account for nodes without a channel (i.e. they are created on the clipboard). Would it be possible to skip this step if there is no channel? We probably don't need to track what happens on users' clipboards since they're private